### PR TITLE
New error behaviour, fixes #5

### LINKED ISSUE #5: Error behaviour
As @mrfishie says:

> In artemis-protocol instead of calling _debugBuffer I provided the buffer as part of the error object, which makes more sense in my opinion. It allows the application to do something else with the buffer data other than just output it (for example, the parent application might want to log to a file, or do something else). Also, from what I can tell, its generally good practice to disconnect the client from the server if a parse error occurs (and for not-so-bad errors you might then want to do something else than trigger the 'error' event, as that event throws the error if it isn't handled).

### PR DESCRIPTION
**Pull request by mrfishie**
Fixes https://github.com/IvanSanchez/node-artemis-lib/issues/5.
Implements two events for error behaviour: one for 'critical' errors such as bad magic number or bad header, the other for not-so-critical errors such as unknown type/subtype, length mismatch, etc.

### CODE REVIEW

File: src/artemisSocket.js [hunk 1]
```diff
@@ -21,7 +17,14 @@ class Socket extends net.Socket {
 
 		this._debug = !!options.debug;
 		this._buffer = null;
-		this.on('data',this._parseData);
+		this.on('data', this._parseData);
+
+		this.on('error', this._onError);
```
Comment (IvanSanchez) [inline src/artemisSocket.js]: I don't think this is needed - the functionality of `net.Socket` can take care of panicking on critical errors, I think.

File: src/artemisSocket.js [hunk 2]
```diff
@@ -100,8 +100,7 @@ class Socket extends net.Socket {
 			try {
 				packet       = packetDefsByType[header.type][subtype].fields.unpack(buffer);
 			} catch(err) {
-				this.emit('error', new ParseError('Could not parse packet (' + header.type.toString(16) + ',' + subtype.toString(16) + '): ' + err.message));
-				this._debugBuffer(buffer);
+				this.emit('error', new ParseError('Could not parse packet (' + header.type.toString(16) + ',' + subtype.toString(16) + '): ' + err.message, buffer));
```
Comment (IvanSanchez) [inline src/artemisSocket.js]: `this.emit('unparsed');` !!!

File: src/artemisSocket.js (hunk 1 above)
Comment (mrfishie) [inline src/artemisSocket.js]: Thanks! Fixed in https://github.com/mrfishie/node-artemis-lib/commit/491cd23cb6eb7fd6c0086cad589deaf8f9673199

**Comment by IvanSanchez** [comment]
:+1:

**Pull request merged**